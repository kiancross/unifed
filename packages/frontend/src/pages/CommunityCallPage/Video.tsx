/*
 * CS3099 Group A3
 */

// Taken some code from
// https://blog.logrocket.com/responsive-camera-component-react-hooks/

import { useRef, useEffect } from "react";

export type VideoProps = { stream?: MediaStream; muted?: boolean };

export function Video(props: VideoProps) {
  const videoRef = useRef<HTMLVideoElement>(null);

  const handleCanPlay = () => {
    if (videoRef.current) {
      videoRef.current.play();
    }
  };

  useEffect(() => {
    if (props.stream && videoRef.current && videoRef.current.srcObject !== props.stream) {
      videoRef.current.srcObject = props.stream;
    }
  }, [videoRef, props.stream]);

  return (
    <video
      ref={videoRef}
      onCanPlay={handleCanPlay}
      autoPlay
      playsInline
      muted={props.muted}
      style={{ width: "100%" }}
    />
  );
}
