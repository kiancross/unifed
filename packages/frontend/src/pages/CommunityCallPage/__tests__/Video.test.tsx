/*
 * CS3099 Group A3
 */

import renderer from "react-test-renderer";
import { Video } from "../Video";

test("Render", () => {
  const component = renderer.create(<Video />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
