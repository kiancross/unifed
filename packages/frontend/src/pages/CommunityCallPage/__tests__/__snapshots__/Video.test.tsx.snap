// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render 1`] = `
<video
  autoPlay={true}
  onCanPlay={[Function]}
  playsInline={true}
  style={
    Object {
      "width": "100%",
    }
  }
/>
`;
